### PR TITLE
fix(files): Open new file when created from template

### LINKED ISSUE #42097: [Bug]: Issues with newly created files (set name and open automatically)
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

In previous versions when creating a new file you could set a name and the new file was opened directly with the default file action.

With 28 it seems that this behaviour changed in multiple ways due to the files to vue migration:

- There is no way to edit the file name directly after selecting the template
- The file is not highlighted anymore
- The new file is not opening automatically anymore

### Steps to reproduce

1. Create a new text file through the new file menu

### Expected behavior

- The file name is editable
- The file entry is highlighted
- The file opens automatically after confirming the new name

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: https://github.com/nextcloud/server/issues/42097

## Summary
Fix a small bug from https://github.com/nextcloud/server/pull/42993 where the, not implemented, `node:focus` event was used instead of changing the route.

### Screencast

https://github.com/nextcloud/server/assets/1855448/32d3ad84-634a-448c-80b0-ebd7ee38df66

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: apps/files/src/views/TemplatePicker.vue [hunk 1]
```diff
@@ -243,7 +243,13 @@ export default defineComponent({
 
 				// Update files list
 				emit('files:node:created', node)
-				emit('files:node:focus', node)
+
+				// Open the new file
+				window.OCP.Files.Router.goToRoute(
+					null, // use default route
+					{ view: 'files', fileid: node.fileid },
+					{ dir: node.dirname, openfile: 'true' },
```
Comment (Pytal) [inline apps/files/src/views/TemplatePicker.vue:251]:
```suggestion
					{ dir: node.dirname, openfile: true },
```
From the code it should be boolean? https://github.com/nextcloud/server/blob/d65e46f31367e65b03d4ab829baeacbece04850a/apps/files/src/components/FilesListVirtual.vue#L212

**Review by Pytal — CHANGES_REQUESTED** [review]

**Review by Pytal — APPROVED** [review]

File: apps/files/src/views/TemplatePicker.vue (hunk 1 above)
Comment (skjnldsv) [reply apps/files/src/views/TemplatePicker.vue:251]: the API doesn't accept boolean :)
But that did trigger an interesting question, and after checking, if you want `&openfile` and not `&openfile=true`, which I thought the boolean would do, you need to use `null` :see_no_evil: 

image

Comment (skjnldsv) [reply apps/files/src/views/TemplatePicker.vue:251]: Also, the openfile watcher is watching the computed, which doesn't care about the value, it just checks if it's there or not ;)
https://github.com/nextcloud/server/blob/8be4704e1125b6eb6fe17a58c8b54c651fa40c0c/apps/files/src/components/FilesListVirtual.vue#L159-L161

**Pull request merged**